lib/Bloc/exercise_bloc.dart

```diff
@@ -4,12 +4,13 @@ import 'package:exerlog/Bloc/user_bloc.dart';
 import 'package:exerlog/Models/exercise.dart';
 import 'package:exerlog/Models/sets.dart';
 
-import '../main.dart';
+import '../src/core/base/shared_preference/shared_preference_b.dart';
+
 
 Future<Exercise> getSpecificExercise(String id) async {
   final data = FirebaseFirestore.instance
       .collection('users')
-      .doc(userID)
+      .doc(SharedPref.getStringAsync('USER_UID'))
```

Comment:
We probably shouldn't be calling the sharedPrefs every single time we use this user uid. Best thing would be to have a singleton user repository that has all the user information. then you can just reference this repository here instead of calling sharedPrefs everytime.

Reply:
Let's tackle this later. On a separate PR.

Reply:
@jorre127 with this repositiory is a good idea, did you have some example?

Reply:
@dinurymomshad Seeing as this pr is about refactoring the use of UUID, might as well do it in this PR no?

Reply:
@mfederowicz Don't know exactly how you do it in riverpods, but it's basically just a singleton you can inject. I think there is already a repository in the app right now called AuthenticationRepository you can look there for inspiration.

Reply:
Since we are using the riverpod, we don't need to use a singleton. Providers once initialized return the same instance every time. So we are good. Implementation example: `dependency.dart`

Reply:
Ok @jorre127 and @dinurymomshad if you dont have further objections, I will try to fix usage of SharedPref.getStringAsync('USER_UID') in another PR?